packages/framework-core/src/services/graphql/websocket-protocol/graphql-websocket-protocol.ts

```diff
@@ -39,34 +47,48 @@ export class GraphQLWebsocketHandler {
 
     try {
       this.logger.debug('Handling websocket message')
+      if ('error' in envelope) {
+        throw envelope.error
+      }
       if (!envelope.value) {
         throw new Error('Received an empty GraphQL body')
       }
+
       const clientMessage = envelope.value as GraphQLClientMessage
       this.logger.debug('Received client message: ', clientMessage)
       switch (clientMessage.type) {
         case MessageTypes.GQL_CONNECTION_INIT:
-          return await this.handleInit(envelope.connectionID)
+          return await this.handleInit(envelope.connectionID, clientMessage)
         case MessageTypes.GQL_START:
           return await this.handleStart(envelope.connectionID, envelope, clientMessage)
         case MessageTypes.GQL_STOP:
           return await this.handleStop(envelope.connectionID, clientMessage)
         case MessageTypes.GQL_CONNECTION_TERMINATE:
           return await this.handleTerminate(envelope.connectionID)
-          break
         default:
           // This branch should be impossible, but just in case
           throw new Error(`Unknown message type. Message=${clientMessage}`)
       }
     } catch (e) {
       this.logger.error(e)
-      await this.sendToConnection(envelope.connectionID, new GraphQLInitError(e.message))
+      await this.connectionManager.sendMessage(this.config, envelope.connectionID, new GraphQLInitError(e.message))
     }
   }
 
-  private async handleInit(connectionID: string): Promise<void> {
+  private async handleInit(connectionID: string, clientMessage: GraphQLInit): Promise<void> {
+    let userEnvelope: UserEnvelope | undefined
+    if (clientMessage.payload?.Authorization) {
+      userEnvelope = await this.authManager.fromAuthToken(clientMessage.payload.Authorization)
+    }
+    const nowEpoch = Math.floor(new Date().getTime() / 1000)
+    const connectionData: ConnectionDataEnvelope = {
+      expirationTime: nowEpoch + this.config.subscriptions.maxConnectionDurationInSeconds,
+      user: userEnvelope,
+    }
+    this.logger.debug('Storing connection data: ', connectionData)
+    await this.connectionManager.storeData(this.config, connectionID, connectionData)
     this.logger.debug('Sending ACK')
-    await this.sendToConnection(connectionID, new GraphQLInitAck())
+    await this.connectionManager.sendMessage(this.config, connectionID, new GraphQLInitAck())
   }
 
   private async handleStart(
```

Comment:
This method is called on every message in the socket. We need to get the user data from the connections store and set it in the envelope we are building to start the GraphQL operation. Check the `augmentEnvelope` method